tools/profileBuilder/latest.go

```diff
@@ -0,0 +1,178 @@
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/marstr/collection"
+)
+
+// LatestStrategy evaluates the Azure-SDK-for-Go repository for the latest available API versions of each service.
+type LatestStrategy struct {
+	Root      string
+	Predicate func(packageName string) bool
+}
+
+// AcceptAll is a predefined value for `LatestStrategy.Predicate` which always returns true.
+func AcceptAll(name string) (result bool) {
+	result = true
+	return
+}
+
+// IgnorePreview searches a packages "API Version" to see if it contains the word "preview". It only returns true when a package is not a preview.
+func IgnorePreview(name string) (result bool) {
+	matches := packageName.FindStringSubmatch(name)
+	if len(matches) >= 4 {
+		result = !strings.Contains(matches[2], "preview")
+	}
+	return
+}
+
+var packageName = regexp.MustCompile(`service[/\\](?P<provider>[\w\-\.\d_]+)[/\\](?:management[/\\])?(?P<version>[\d\-\w\._]+)[/\\](?P<group>[\w\d\-\._]+)`)
+
+// Enumerate scans through the known Azure SDK for Go packages and finds each
+func (latest LatestStrategy) Enumerate(cancel <-chan struct{}) collection.Enumerator {
+	results := make(chan interface{})
+
+	go func() {
+		defer close(results)
+
+		type operationGroup struct {
+			provider     string
+			resourceType string
+			group        string
+		}
+
+		type operInfo struct {
+			version string
+			rawpath string
+		}
+
+		maxFound := make(map[operationGroup]operInfo)
+
+		filepath.Walk(latest.Root, func(currentPath string, info os.FileInfo, openErr error) (err error) {
+			pathMatches := packageName.FindStringSubmatch(currentPath)
+
+			if latest.Predicate == nil {
+				latest.Predicate = AcceptAll
+			}
+
+			if len(pathMatches) == 0 || !info.IsDir() || !latest.Predicate(currentPath) {
+				return
+			}
+
+			version := pathMatches[2]
+			currentGroup := operationGroup{
+				provider:     pathMatches[1],
+				resourceType: pathMatches[2],
+				group:        pathMatches[3],
+			}
+
+			prev, ok := maxFound[currentGroup]
+			if !ok {
+				maxFound[currentGroup] = operInfo{version, currentPath}
+				return
+			}
+
+			if le, _ := versionle(prev.version, version); le {
+				maxFound[currentGroup] = operInfo{version, currentPath}
+			}
+
+			return
+		})
+
+		for _, entry := range maxFound {
+			absolute, err := filepath.Abs(entry.rawpath)
+			if err != nil {
+				continue
+			}
+
+			select {
+			case results <- absolute:
+				// Intionally Left Blank
+			case <-cancel:
+				return
+			}
+		}
+	}()
+
+	return results
+}
+
+// ErrNotVersionString is instantiated when a string not conforming to known Azure API Version patterns is parsed is if it did.
+type ErrNotVersionString string
+
+func (err ErrNotVersionString) Error() string {
+	return fmt.Sprintf("`%s` is not a recognized Azure version string", string(err))
+}
+
+var versionle = func() func(string, string) (bool, error) {
+	versionPattern := regexp.MustCompile(`^(?P<year>\d{4})-(?P<month>\d{2})-(?P<day>\d{2})(?:[\.\-](?P<tag>.+))?$`)
```

Comment:
>versionPattern [](start = 1, length = 14)

Graphrbac uses a different pattern

Reply:
I think the amount of work for this versus the reward warrants fixing this in a separate PR.